### ISSUE TITLE
QueryCursor.js is throwing an error on line 246: "TypeError: Cannot read property 'length' of undefined" 

### ISSUE THREAD

**Issue by bam-tbf** [Automattic/mongoose#7169]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

QueryCursor.js is throwing an error on line 246:
```
  if (ctx._transforms.length) {
```
Trying to find the length of an undefined property (ctx._transforms).

```
11:14:02 🚨 Unhandled promise rejection at: Promise [object Promise] Reason: TypeError: Cannot read property 'length' of undefined   Trace: TypeError: Cannot read property 'length' of undefined
    at _next (/Users/user/code/its/node_modules/mongoose/lib/cursor/QueryCursor.js:246:23)
    at QueryCursor._read (/Users/user/code/its/node_modules/mongoose/lib/cursor/QueryCursor.js:70:3)
    at QueryCursor.Readable.read (_stream_readable.js:442:10)
    at flow (_stream_readable.js:846:34)
    at Stream.<anonymous> (_stream_readable.js:707:7)
    at emitNone (events.js:106:13)
    at Stream.emit (events.js:208:7)
    at Stream._resume (/Users/user/code/its/node_modules/highland/lib/index.js:1182:22)
    at Stream._checkBackPressure (/Users/user/code/its/node_modules/highland/lib/index.js:1091:10)
    at Stream._resume (/Users/user/code/its/node_modules/highland/lib/index.js:1172:33)
    at Stream._checkBackPressure (/Users/user/code/its/node_modules/highland/lib/index.js:1091:10)
    at Stream._resume (/Users/user/code/its/node_modules/highland/lib/index.js:1172:33)
    at Stream.resume (/Users/user/code/its/node_modules/highland/lib/index.js:1205:10)
    at Stream.pull (/Users/user/code/its/node_modules/highland/lib/index.js:1596:7)
    at Stream._generator (/Users/user/code/its/node_modules/highland/lib/index.js:3672:20)
    at Stream._runGenerator (/Users/user/code/its/node_modules/highland/lib/index.js:1343:10)
    at Stream._resume (/Users/user/code/its/node_modules/highland/lib/index.js:1178:22)
    at Stream._checkBackPressure (/Users/user/code/its/node_modules/highland/lib/index.js:1091:10)
    at Stream._resume (/Users/user/code/its/node_modules/highland/lib/index.js:1172:33)
    at Stream._checkBackPressure (/Users/user/code/its/node_modules/highland/lib/index.js:1091:10)
    at Stream._resume (/Users/user/code/its/node_modules/highland/lib/index.js:1172:33)
    at Stream.resume (/Users/user/code/its/node_modules/highland/lib/index.js:1205:10) 

```

**If the current behavior is a bug, please provide the steps to reproduce.**

This bug only arose when I upgrade to 5.3.5. I don't think I can easily provide a script to replicate. Hopefully it is not needed because it seems there is an internal error check missing to assure ctx._transforms is not undefined.

I upgraded to 5.3.5 to take advantage of the fix for replaceOne that was a part of this release.

**What is the expected behavior?**

QueryCursor.js should not throw an error.

**Please mention your node.js, mongoose and MongoDB version.**

mongoose: 5.3.5
node: 8.9.2
mongodb: 3.6.8

**Comment by gaohuifeng** [comment]
same problem,
code
```
const cursor = db.task
      .find(conds)
      .select('_id uniqueId')
      .lean()
      .cursor()

cursor.next()  // error： TypeError: Cannot read property 'length' of undefined， at _next (/Users/ghf/workspace/core/node_modules/mongoose/lib/cursor/QueryCursor.js:246:23)
```

**Comment by bam-tbf** [comment]
Confirming this bug was fixed by release 5.3.6. @gaohuifeng

**Comment by lineus** [comment]
@bam-tbf thanks for reporting it @bam-tbf. You're correct, this should be fixed as of @5.3.6.
